tests/test_modeling_flax_vision_encoder_decoder.py

```diff
@@ -0,0 +1,339 @@
+        self.check_encoder_decoder_model_generate(**config_inputs_dict)
+
+    def test_real_model_save_load_from_pretrained(self):
+        model_2 = self.get_pretrained_model()
+        pixel_values = floats_tensor(
+            [
+                13,
+                model_2.config.encoder.num_channels,
+                model_2.config.encoder.image_size,
+                model_2.config.encoder.image_size,
+            ]
+        )
+        decoder_input_ids = ids_tensor([13, 1], model_2.config.decoder.vocab_size)
+
+        outputs = model_2(
+            pixel_values=pixel_values,
+            decoder_input_ids=decoder_input_ids,
+        )
+        out_2 = np.array(outputs[0])
+        out_2[np.isnan(out_2)] = 0
+
+        with tempfile.TemporaryDirectory() as tmp_dirname:
+            model_2.save_pretrained(tmp_dirname)
+            model_1 = FlaxVisionEncoderDecoderModel.from_pretrained(tmp_dirname)
+
+            after_outputs = model_1(
+                pixel_values=pixel_values,
+                decoder_input_ids=decoder_input_ids,
+            )
+            out_1 = np.array(after_outputs[0])
+            out_1[np.isnan(out_1)] = 0
+            max_diff = np.amax(np.abs(out_1 - out_2))
+            self.assertLessEqual(max_diff, 1e-5)
+
+
+@require_flax
+class FlaxVisionGPT2EncoderDecoderModelTest(FlaxVisionEncoderDecoderMixin, unittest.TestCase):
+    def get_encoder_decoder_model(self, config, decoder_config):
+        encoder_model = FlaxViTModel(config)
+        decoder_model = FlaxGPT2LMHeadModel(decoder_config)
+        return encoder_model, decoder_model
+
+    def prepare_config_and_inputs(self):
+        model_tester_encoder = FlaxViTModelTester(self, batch_size=13)
+        model_tester_decoder = FlaxGPT2ModelTester(self, batch_size=13)
+        encoder_config_and_inputs = model_tester_encoder.prepare_config_and_inputs()
+        decoder_config_and_inputs = model_tester_decoder.prepare_config_and_inputs_for_decoder()
+        (config, pixel_values) = encoder_config_and_inputs
+        (
+            decoder_config,
+            decoder_input_ids,
+            decoder_attention_mask,
+            encoder_hidden_states,
+            encoder_attention_mask,
+        ) = decoder_config_and_inputs
+
+        # make sure that cross attention layers are added
+        decoder_config.add_cross_attention = True
+        return {
+            "config": config,
+            "pixel_values": pixel_values,
+            "decoder_config": decoder_config,
+            "decoder_input_ids": decoder_input_ids,
+            "decoder_attention_mask": decoder_attention_mask,
+            "encoder_hidden_states": encoder_hidden_states,  # This is not used in the tests.
+        }
+
+    def get_pretrained_model(self):
+        return FlaxVisionEncoderDecoderModel.from_encoder_decoder_pretrained(
+            "google/vit-base-patch16-224-in21k", "gpt2"
+        )
+
+
+@require_flax
+class FlaxVisionEncoderDecoderModelTest(unittest.TestCase):
```

Comment:
Could we maybe add one "real" integration test for say image captioning which shows that the model works correctly? We could use your image captioning model for French if you'd like :-)

Reply:
Hey @patrickvonplaten,

I got TRC TPU access yesterday (for 3 months). I would prefer to train a `FlaxVisionEncoderDecoderModel` model using this resource :). The French model has 2 major bugs and gave very bad result. Also using it requires dealing with the weight loading things.

I think I can launch the training next week.

Reply:
Let me change to WIP for now.